packages/mobile/src/Components/Button.tsx

```diff
@@ -130,6 +139,21 @@ const Button: React.FunctionComponent<Omit<ButtonProps, 'height' | 'width'>> = (
     style = {};
   }
 
+  if (props.icon) {
```

Comment:
This is just a commentary, not a request for changes. When you have mutable properties that don't need to be in the state, use the `useMemo` or `useCallback` hooks depending on the case.

On line 109, you could do:
```js
const margin = React.useCallback(() => {
  calculate_the_margin_here_and_return_it
},[props.icon]);
```

this will recalculate the value of `margin` only when the value of `props.icon` changes, not on all changes on the props